### PR TITLE
Document ARM support

### PR DESCRIPTION
**Pull request by MrAlias**
Resolve https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/132

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,12 @@ Currently, OpenTelemetry Go Automatic Instrumentation is tested for the followin
 | Ubuntu  | 1.18       | amd64        |
 
 Automatic instrumentation should work on any Linux kernel above 4.4.
-Windows/Mac users should currently use Docker/VM to compile and run this repository.
+
+This project supports the ARM platform.
+However, there is no automated testing for this platform.
+Be sure to validate support on your own ARM bases system.
```
Comment (JamieDanielson) [inline README.md:33]:
```suggestion
Be sure to validate support on your own ARM based system.
```

**Review by JamieDanielson — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -27,7 +27,12 @@ Currently, OpenTelemetry Go Automatic Instrumentation is tested for the followin
 | Ubuntu  | 1.18       | amd64        |
 
 Automatic instrumentation should work on any Linux kernel above 4.4.
-Windows/Mac users should currently use Docker/VM to compile and run this repository.
+
+This project supports the ARM platform.
+However, there is no automated testing for this platform.
+Be sure to validate support on your own ARM based system.
+
+Windows users should currently use Docker/VM to compile and run this repository.
```
Comment (pellared) [inline README.md:35]: This is not only about Windows 😉 

```suggestion
Users of non-Linux operating systems can use
[the Docker images](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pkgs/container/opentelemetry-go-instrumentation%2Fautoinstrumentation-go)
or create a virtual machine to compile and run OpenTelemetry Go Automatic Instrumentation.
```

File: README.md [hunk 3]
```diff
@@ -27,7 +27,12 @@ Currently, OpenTelemetry Go Automatic Instrumentation is tested for the followin
 | Ubuntu  | 1.18       | amd64        |
 
 Automatic instrumentation should work on any Linux kernel above 4.4.
-Windows/Mac users should currently use Docker/VM to compile and run this repository.
+
+This project supports the ARM platform.
```
Comment (pellared) [inline README.md:31]:
```suggestion
OpenTelemetry Go Automatic Instrumentation supports the arm64 architecture.
```

arm64 is the "key" change.

**Review by pellared — CHANGES_REQUESTED** [review]

**Review by MikeGoldsmith — APPROVED** [review]
Pending @pellared's review comments, this looks good to me 👍🏻

**Pull request merged**